python/lib/sift_py/ingestion/_internal/ingest.py

```diff
@@ -58,13 +59,17 @@ def __init__(
     ):
         ingestion_config = self.__class__._get_or_create_ingestion_config(channel, config)
 
-        if not overwrite_rules:
-            self.__class__._validate_rules_synchronized(
-                channel, ingestion_config.asset_id, config.rules
+        if overwrite_rules:
+            warn(
+                "The 'overwrite_rules' argument is deprecated and will be removed in a future release."
             )
 
+        self.rule_service = RuleService(channel)
         if len(config.rules) > 0:
-            update_rules(channel, ingestion_config.asset_id, config.rules)
+            for rule in config.rules:
+                if config.asset_name not in rule.asset_names:
```

Comment:
I felt like part of the simplicity of just including rules in the telemetry config was so that you didn't have to necessarily separately configure the asset_names list on the rules, when the telemetry config they were a part of is already tied to a specific asset. So I thought it'd make sense to just ensure that the asset is configured on the rule by default. Let me know if you think that makes sense though.

Reply:
makes sense to me!